Detectors/TPC/workflow/include/TPCWorkflow/TPCAverageGroupIDCSpec.h

```diff
@@ -0,0 +1,119 @@
+    : mLane{lane}, mCRUs{crus}, mInputFormat{inputFormat}, mDebug{debug}
+  {
+    auto& paramIDCGroup = ParameterIDCGroup::Instance();
+    for (const auto& cru : mCRUs) {
+      const int reg = cru % Mapper::NREGIONS;
+      mIDCs.emplace(cru, IDCAverageGroup(paramIDCGroup.GroupPads[reg], paramIDCGroup.GroupRows[reg], paramIDCGroup.GroupLastRowsThreshold[reg], paramIDCGroup.GroupLastPadsThreshold[reg], reg));
+    }
+  }
+
+  void run(o2::framework::ProcessingContext& pc) final
+  {
+    LOGP(info, "averaging and grouping IDCs for one TF for CRUs {} to {} using {} threads", mCRUs.front(), mCRUs.back(), mIDCs.begin()->second.getNThreads());
+
+    for (int i = 0; i < mCRUs.size(); ++i) {
+      const DataRef ref = pc.inputs().getByPos(i);
+      auto const* tpcCRUHeader = o2::framework::DataRefUtils::getHeader<o2::header::DataHeader*>(ref);
+      const int cru = tpcCRUHeader->subSpecification >> 7;
+      mIDCs[cru].setIDCs(pc.inputs().get<std::vector<float>>(ref));
+      mIDCs[cru].processIDCs();
+
+      if (mDebug) {
+        mIDCs[cru].dumpToFile(fmt::format("IDCGroup_{}.root", mLane).data(), fmt::format("CRU_{}_tf_{}", cru, tpcCRUHeader->tfCounter).data());
+      }
+
+      // send the output for one CRU for one TF
+      sendOutput(pc.outputs(), cru);
+    }
+  }
+
+  void endOfStream(o2::framework::EndOfStreamContext& ec) final
+  {
+    LOGP(info, "endOfStream");
+    ec.services().get<ControlService>().readyToQuit(QuitRequest::Me);
+  }
+
+ private:
+  const int mLane{0};                                        ///< lane number of processor
+  const std::vector<uint32_t> mCRUs{};                       ///< CRUs to process in this instance
+  const TPCIntegrateIDCDevice::IDCFormat mInputFormat{};     ///< type of the input format. Sim=simulation, Real=realistic format
+  const bool mDebug{};                                       ///< dump IDCs to tree for debugging
+  std::unordered_map<unsigned int, IDCAverageGroup> mIDCs{}; ///< object for averaging and grouping the IDCs
+
+  void sendOutput(DataAllocator& output, const uint32_t cru)
+  {
+    const header::DataHeader::SubSpecificationType subSpec{cru << 7};
+    output.snapshot(Output{gDataOriginTPC, "IDCGROUP", subSpec, Lifetime::Timeframe}, mIDCs[cru].getIDCGroup().getData());
+  }
+};
+
+DataProcessorSpec getTPCAverageGroupIDCSpec(const int ilane = 0, const std::vector<uint32_t>& crus = {}, const TPCIntegrateIDCDevice::IDCFormat inputFormat = {}, const bool debug = false)
+{
+  std::vector<OutputSpec> outputSpecs;
+  std::vector<InputSpec> inputSpecs;
+  outputSpecs.reserve(crus.size());
+  inputSpecs.reserve(crus.size());
+
+  for (const auto& cru : crus) {
+    const header::DataHeader::SubSpecificationType subSpec{cru << 7};
+    inputSpecs.emplace_back(InputSpec{"idcs", gDataOriginTPC, TPCIntegrateIDCDevice::getDataDescription(inputFormat), subSpec, Lifetime::Timeframe});
```

Comment:
I prepared a device for decoding the IDC here: https://github.com/AliceO2Group/AliceO2/pull/6136. The output for the moment I call `ICDVECTOR` which should be identical to your simulation output I think. If we agree on a name, then it could already work. The only thing to check, from the CRU the IDCs in one TF, so one snapshot call, can come from either 12 or 13 TFs. The size from my objects is always for 13 TFs for the moment. I send a second object which contains the orbit information of all IDCs in the TF. This will be needed on the merger I think to check all data arrives.